hedera-node/hedera-app/src/main/java/com/hedera/node/app/services/ServicesRegistryImpl.java

```diff
@@ -65,10 +66,14 @@ public ServicesRegistryImpl(
     /**
      * Register the given service.
      *
-     * @param service The service to register
+     * @param schemaAware The implementation to register
      */
     @Override
-    public void register(@NonNull final Service service) {
+    public void register(@NonNull final SchemaAware schemaAware) {
+        requireNonNull(schemaAware);
+        if (!(schemaAware instanceof final Service service)) {
```

Comment:
Why though? There is nothing that prevents this class from handling your new class. It should probably be called `SchemaAwareRegistryImpl` and its interface should be renamed as well. 

Now you see the challenge with renaming of the interface `Service` into `SchemaAware`. There multiple implicit dependencies you discover only after you try this renaming.

Reply:
it does other things at the service level though which i don't want to be visible to (lower level) schema aware.
I didn't set out to rename `Service` to `SchemaAware`, but to extract a lower level `SchemaAware` out of it. 
This option is minimally invasive, and achieves the same purpose. The instanceOf isn't ideal but it's better than casting.

Reply:
My point is that this is the only spot in the code where `MerkleSchemaRegistry` is created. 

Here is how the callstack goes:
- `ServicesRegistry#register` in `com.hedera.node.app.Hedera#Hedera`. 
- `SchemaAware#registerSchemas` in `com.hedera.node.app.services.ServicesRegistryImpl#register`
- `SchemaRegistry#register` in `SchemaAware#registerSchemas`. And this is where an instance of a schema finally gets in the registry 

If you have this condition in place
```
   if (!(schemaAware instanceof final Service service)) {
```
your schema is not going to be registered.

Reply:
Thanks. On a second look, even if i fixed this issue you pointed (thank you), there's much more going on with the classId generation using the service name (which i'd have to stop providing) in `StateUtils`. I conclude this "only services use schemas" assumption is deep-rooted, and can't be removed trivially.

        // NOTE: Once this is live on any network, the formula used to generate this key can NEVER
        // BE CHANGED or you won't ever be able to deserialize an exising state! If we get away from
        // this formula, we will need to hardcode known classId that had been previously generated.

I'd rather avoid this work. I'll have to create a dummy service :(

Reply:
I don't think we need to introduce a SchemaAware. We have an example of a service that doesn't have a public API in the BlockRecordService.